be/src/service/staros_worker.cpp

```diff
@@ -122,7 +132,14 @@ absl::Status StarOSWorker::invalidate_fs(const ShardInfo& info) {
 
 absl::Status StarOSWorker::remove_shard(const ShardId id) {
     std::unique_lock l(_mtx);
-    _shards.erase(id);
+    auto iter = _shards.find(id);
+    if (iter != _shards.end()) {
+#ifndef BE_TEST
+        uint64_t table_id = get_table_id(iter->second.shard_info);
+        StarRocksMetrics::instance()->table_metrics_mgr()->unregister_table(table_id);
```

Comment:
what if the "tableId" is not there in the shard property?

Reply:
BE will crash. The existence of tableId is a rule that must be followed in the shared data mode. At least currently, it seems that we have no other way to obtain tableId from the shard.